tensorflow/compiler/tf2tensorrt/convert/ops/layer_utils.h

```diff
@@ -0,0 +1,377 @@
+    StatusOr<nvinfer1::IElementWiseLayer*> sign = this->SignInt(input);
+    return this->Mul(input, (*sign)->getOutput(0));
+  }
+
+  // Returns elementwise sign(x) for int32 input tensors where sign(x) is
+  // defined as 1 where x > 0, -1 where x < 0 and 0 where x == 0.
+  StatusOr<nvinfer1::IElementWiseLayer*> SignInt(
+      nvinfer1::ITensor* input) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(input);
+
+    // Create constants +1 and -1.
+    StatusOr<nvinfer1::IConstantLayer*> one =
+        this->Constant<int32>(1, input->getDimensions().nbDims);
+    TRT_ENSURE_PTR_OK(one);
+
+    StatusOr<nvinfer1::IConstantLayer*> neg_one =
+        this->Constant<int32>(-1, input->getDimensions().nbDims);
+    TRT_ENSURE_PTR_OK(neg_one);
+
+    // Make all negaitve elements into -1, positive and zero elements
+    // unaffected.
+    StatusOr<nvinfer1::IElementWiseLayer*> max =
+        this->Max(input, (*neg_one)->getOutput(0));
+    TRT_ENSURE_PTR_OK(max);
+
+    // Make all positive elements into +1, negative and zero elements
+    // unaffected.
+    StatusOr<nvinfer1::IElementWiseLayer*> min =
+        this->Min((*max)->getOutput(0), (*one)->getOutput(0));
+    TRT_ENSURE_PTR_OK(min);
+    return min;
+  }
+
+  // Adds an Add operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Sub(nvinfer1::ITensor* lhs,
+                                             nvinfer1::ITensor* rhs) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(lhs);
+    TRT_ENSURE(rhs);
+    nvinfer1::IElementWiseLayer* layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kSUB);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Adds an Greater operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Greater(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(lhs);
+    TRT_ENSURE(rhs);
+    nvinfer1::IElementWiseLayer* layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kGREATER);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Adds an Equal operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Equal(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(lhs);
+    TRT_ENSURE(rhs);
+    nvinfer1::IElementWiseLayer* layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kEQUAL);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Simulates the creation of a tensor of the same shape as "input", but filled
+  // with 0's where "input" is non-zero. TensorRT doesn't have dynamic creation
+  // functions, so this function multiplies the input by a zero value to side
+  // step knowing the shape. For static inputs, this can be optimized using one
+  // of the TRTNetworkBuilder::Constant function instead.
+  StatusOr<nvinfer1::IElementWiseLayer*> ZerosLikeDyamic(
```

Comment:
I couldn't find any use of the routine.
Dynamic is mis-spelled as Dyamic.

This routine get a tensor with the same shape as the input and with value 0. It is hard to correlate its name ZerosLikeDynamic with its function. Maybe something like "ZerosWithShape"?